bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,557 @@
+                                    }
+                                }
+                                properties.put("Supported fan levels", fanLevels.toString());
+                                updateProperties(properties);
+                                channelId = CHANNEL_TYPE_FANSPEED;
+                                addChannel(channelId, itemType, fanLevels);
+                                break;
+                            case 5:
+                                List<String> swingUDModes = new ArrayList<>();
+                                String[] swingUDString = descr.states;
+                                for (int i1 = 0; i1 < swingUDString.length; i1++) {
+                                    if ("AUTO".contentEquals(swingUDString[i1])) {
+                                        swingUDModes.add("AUTO");
+                                    } else if ("10".contentEquals(swingUDString[i1])) {
+                                        swingUDModes.add("SWING");
+                                    } else if ("11".contentEquals(swingUDString[i1])) {
+                                        swingUDModes.add("SWIRL");
+                                    } else if ("12".contentEquals(swingUDString[i1])) {
+                                        swingUDModes.add("WIDE");
+                                    } else {
+                                        swingUDModes.add(swingUDString[i1]);
+                                    }
+                                }
+                                channelId = CHANNEL_TYPE_VANESUD;
+                                properties.put("Supported vane up/down modes", swingUDModes.toString());
+                                updateProperties(properties);
+                                addChannel(channelId, itemType, swingUDModes);
+                                break;
+                            case 6:
+                                List<String> swingLRModes = new ArrayList<>();
+                                String[] swingLRString = descr.states;
+                                for (int i1 = 0; i1 < swingLRString.length; i1++) {
+                                    if ("AUTO".contentEquals(swingLRString[i1])) {
+                                        swingLRModes.add("AUTO");
+                                    } else if ("10".contentEquals(swingLRString[i1])) {
+                                        swingLRModes.add("SWING");
+                                    } else if ("11".contentEquals(swingLRString[i1])) {
+                                        swingLRModes.add("SWIRL");
+                                    } else if ("12".contentEquals(swingLRString[i1])) {
+                                        swingLRModes.add("WIDE");
+                                    } else {
+                                        swingLRModes.add(swingLRString[i1]);
+                                    }
+                                }
```

Comment:
It looks like this code is almost identical to the code above. Can you refactor this code so it isn't duplicated?

Reply:
Refactored